R/trace_generation.R

```diff
@@ -41,13 +41,24 @@ layer2traces <- function(l, d, misc) {
   
   ## For non-numeric data on the axes, we should take the values from
   ## the original data.
-  for (axis.name in c("x", "y")){
-    if (!misc$is.continuous[[axis.name]]){
+  for (axis.name in c("x", "y")) {
+    if (!misc$is.continuous[[axis.name]]) {
       aes.names <- paste0(axis.name, c("", "end", "min", "max"))
       aes.used <- aes.names[aes.names %in% names(g$aes)]
       for(a in aes.used) {
         col.name <- g$aes[aes.used]
-        data.vec <- l$data[[col.name]]
+        dtemp <- l$data[[col.name]]
+        if (is.null(dtemp)) {
+          if (!inherits(g$data[[paste0(a, ".name")]], "NULL")) {
+            # Handle the case where as.Date() is passed in aes argument.
+            if (class(g$data[[a]]) != class(g$data[[paste0(a, ".name")]])) {
+              g$data[[a]] <- g$data[[paste0(a, ".name")]]
+              data.vec <- g$data[[a]]
+            }
+          }
```

Comment:
There are two if statements without an else. is something bad going to happen if `data.vec` isn't set?